### PR TITLE
NAS-100502: Finish initial draft of ACME documentation

### PR DESCRIPTION
**Pull request by Mrt134**
- TODO: add an example of a working ACME cert, review ported field descriptions from UI help text, expand descriptive elements where needed.
- Add screenshots, background text, and descriptive text of the new ACME options in System/Certificates and System/ACME DNS.
- HTML build test: no issues.

### CODE REVIEW

File: userguide/system.rst [hunk 1]
```diff
@@ -2670,6 +2677,94 @@ Clicking |ui-options| for an entry shows these configuration buttons:
...
+
+.. _ACME Certificate Options:
+
+.. table:: ACME Certificate Options
+   :class: longtable
+
+   +--------------------------------------+----------------+-------------------------------------------------------------------------------------+
+   | Setting                              | Value          | Description                                                                         |
+   +======================================+================+=====================================================================================+
+   | identifier                           | string         | Internal identifier of the certificate. Only alphanumeric, "_" and "-" are allowed. |
```
Comment (wblock) [inline userguide/system.rst]: Spell out punctuation characters in addition to showing them.
```suggestion
   | identifier                           | string         | Internal identifier of the certificate. Only alphanumeric characters, dash (:literal:`-`) and underline (:literal:`_`) are allowed. |
```

File: userguide/system.rst [hunk 2]
```diff
@@ -2670,6 +2677,94 @@ Clicking |ui-options| for an entry shows these configuration buttons:
...
+   +--------------------------------------+----------------+-------------------------------------------------------------------------------------+
+   | Terms of Service                     | checkbox       | Please accept terms of service for the given ACME Server.                           |
+   +--------------------------------------+----------------+-------------------------------------------------------------------------------------+
+   | Renew certificate day                | integer        | Number of days to renew certificate before expiring                                 |
+   +--------------------------------------+----------------+-------------------------------------------------------------------------------------+
+   | ACME Server Directory URI            | drop-down menu | Please specify URI of ACME Server Directory.                                        |
```
Comment (wblock) [inline userguide/system.rst]: This one seems out of character with the others:
```suggestion
   | ACME Server Directory URI            | drop-down menu | URI of the ACME Server Directory.                                        |
```

File: userguide/system.rst [hunk 3]
```diff
@@ -2670,6 +2677,94 @@ Clicking |ui-options| for an entry shows these configuration buttons:
...
+   +--------------------------------------+----------------+-------------------------------------------------------------------------------------+
+   | Authenticator for :samp:`{hostname}` | drop-down menu | Specify Authenticator to be used for :samp:`{hostname}`                             |
```
Comment (wblock) [inline userguide/system.rst]: "hostname" is not clear here--is it a field name or contents? Where is it found or entered? Also, "specify" is not needed. "to be used" is future tense.
```suggestion
   | Authenticator for :samp:`{hostname}` | drop-down menu | Authenticator to use for :samp:`{hostname}`                             |
```

File: userguide/system.rst [hunk 4]
```diff
@@ -2670,6 +2677,94 @@ Clicking |ui-options| for an entry shows these configuration buttons:
...
+   +--------------------------------------+----------------+-------------------------------------------------------------------------------------+
+
+
+.. index:: ACME DNS
+.. _ACME DNS:
+
+ACME DNS
+--------
+
+Going to
```
Comment (wblock) [inline userguide/system.rst]: This is past tense.
```suggestion
Go to
```

File: userguide/system.rst [hunk 5]
```diff
@@ -2670,6 +2677,94 @@ Clicking |ui-options| for an entry shows these configuration buttons:
...
+:menuselection:`System --> ACME DNS`
+and clicking :guilabel:`ADD` shows options to add a new DNS
```
Comment (wblock) [inline userguide/system.rst]:
```suggestion
and click :guilabel:`ADD` to show options to add a new DNS
```

File: userguide/system.rst [hunk 6]
```diff
@@ -2670,6 +2677,94 @@ Clicking |ui-options| for an entry shows these configuration buttons:
...
+  account :guilabel:`Access ID Key` and :guilabel:`Secret Access Key`.
+  See the
+  `AWS documentation <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html>`__
+  for more details about generating these keys.
+
+* **Hover:** `Commercial DNS Provider <https://www.hover.com/>`__. No
+  additional attributes are required.
+
+
+Clicking :guilabel:`SAVE` registers the DNS Authenticator and adds it to
```
Comment (wblock) [inline userguide/system.rst]:
```suggestion
Click :guilabel:`SAVE` to register the DNS Authenticator and add it to
```

**Comment by Mrt134** [comment]
webui PR: https://github.com/freenas/webui/pull/2637

**Pull request merged**